test/models/workflow_execution_test.rb

```diff
@@ -94,4 +94,173 @@ def setup
     assert @workflow_execution_valid.canceled?
   end
+
+  test 'cancellable' do
+    @workflow_execution_valid.state = :initial
+    assert @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :submitted
+    assert @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :running
+    assert @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :prepared
+    assert @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :completing
+    assert_not @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :completed
+    assert_not @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :error
+    assert_not @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :canceling
+    assert_not @workflow_execution_valid.cancellable?
+
+    @workflow_execution_valid.state = :canceled
+    assert_not @workflow_execution_valid.cancellable?
+  end
+
+  test 'deletable' do
+    @workflow_execution_valid.state = :completed
+    assert @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :error
+    assert @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :canceled
+    assert @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :initial
+    assert_not @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :submitted
+    assert_not @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :running
+    assert_not @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :prepared
+    assert_not @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :completing
+    assert_not @workflow_execution_valid.deletable?
+
+    @workflow_execution_valid.state = :canceling
+    assert_not @workflow_execution_valid.deletable?
+  end
+
+  test 'sent_to_ga4gh' do
+    @workflow_execution_valid.state = :submitted
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :running
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :completing
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :completed
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :error
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :canceling
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :canceled
+    assert @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :prepared
+    assert_not @workflow_execution_valid.sent_to_ga4gh?
+
+    @workflow_execution_valid.state = :initial
+    assert_not @workflow_execution_valid.sent_to_ga4gh?
+  end
+
+  test 'send_email' do
```

Comment:
**nitpick (blocking):** If I run this test case it reports no assertions have been made. If I refactor lines 190 - 194 with the following I actually see assertions.

```ruby
assert_enqueued_emails 1 do
      workflow_execution_completed.send_email
      assert_enqueued_email_with PipelineMailer, :complete_email, args: [workflow_execution_completed]
    end
```

Note: this example will fail as the fixture you are using for this test does not have `email_notification: true`.

Can you refactor all of these case in your test. The issue is that `workflow_execution_completed.send_email do` is not valid syntax. That would only work if the method accepted a block but then again you wouldn't pass test assertions to a block.

Reply:
Thanks for catching this. Fixed in 9eaca37fe764f2a38bec728a104ef44e632e25b5

Reply:
I realized it doesn't make much sense to have `assert_enqueued_email_with PipelineMailer, :complete_email, args: [workflow_execution_example]` in blocks that have 0 enqueued emails. Removed in 00276fea505da70d2136e5c087deb906a3f2f6bd